containers/daap-python-base/src/var/task/curated_data/curated_data_loader.py

```diff
@@ -80,3 +81,182 @@ def ingest_raw_data(self, raw_data_table: QueryTable, extraction_timestamp: str)
             database_name=self.curated_data_table.database,
             table_name=self.curated_data_table.name,
         )
+
+    def create_new_major_version_data_product_table(self, is_updated_table):
+        """
+        Creates new data product version tables
+
+        As the updated table (cause of the verison increase) is already created
+        it uses an UNLOAD rather than CTAS query
+        """
+        if not is_updated_table:
+            qid = start_query_execution_and_wait(
+                database_name=self.curated_data_table.database,
+                sql=self.query_builder.sql_create_next_major_increment_table(
+                    self.curated_data_table,
+                ),
+                logger=self.logger,
+            )
+            self.logger.info(f"created {self.curated_data_table}, using query id {qid}")
+        else:
+            qid = start_query_execution_and_wait(
+                database_name=self.curated_data_table.database,
+                sql=self.query_builder.sql_unload_for_major_updated_table(
+                    curated_table=self.curated_data_table,
+                ),
+                logger=self.logger,
+            )
+            self.logger.info(
+                f"unloaded {self.curated_data_table}, using query id {qid}"
+            )
+            refresh_table_partitions(
+                database_name=self.curated_data_table.database,
+                table_name=self.curated_data_table.name,
+            )
+
+
+class CuratedDataCopier:
+    def __init__(
+        self,
+        column_changes,
+        new_schema: DataProductSchema,
+        element: DataProductElement,
+        athena_client,
+        glue_client,
+        logger=DataPlatformLogger,
+    ):
+        """
+        Copy data from existing version of data product to new version of
+        data product.
+        """
+        self.data_product_name = new_schema.data_product_name
+        self.column_changes = column_changes
+        self.schema = new_schema
+        self.element = element
+        self.new_curated_data_product_path = element.curated_data_prefix.uri
+        self.new_database_name = element.database_name
+        self.glue_client = glue_client
+        self.athena_client = athena_client
+        self.logger = logger
+        self.get_tables_to_copy()
+
+    def validate(self):
```

Comment:
should this be called by the version class? (before `run()`)

Reply:
yeah potentially, could be called from there and then the result used to inform the requester of the copy behaviour